lib/storage/src/chunks/getDataReadableStream.ts

```diff
@@ -0,0 +1,20 @@
+import { Buffer } from "buffer";
+
+export async function* getNextData(data: ReadableStream): AsyncGenerator<Buffer> {
+  // Get a lock on the stream.
+  const reader = data.getReader();
+
+  try {
+    while (true) {
+      // Read from the stream.
+      const { done, value } = await reader.read();
+      // Exit if we're done.
+      if (done) return;
+      // Else yield the chunk.
+      yield Buffer.from(value);
+    }
+  } finally {
```

Comment:
I dont think the catch block is needed since this would be an invalid readableStream and caught a level higher (in the chunker). But i'll add it anyways. I agree it's probably best for this function do that.

Reply:
done